### PR TITLE
Remove deprecated `sync_batchnorm` and `num_nodes` attributes in `DDPSpawnPlugin`

### PR DESCRIPTION
**Pull request by puhuk**
## What does this PR do?

Part of #10312

test_v1_6_0_ddp_spawn_num_nodes()
test_v1_6_0_ddp_spawn_sync_batchnorm()

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 1]
```diff
@@ -83,18 +81,7 @@ def __init__(
             cluster_environment=cluster_environment,
             checkpoint_io=checkpoint_io,
         )
-        if num_nodes is not None:
-            rank_zero_deprecation(
-                "Argument `num_nodes` in `DDPSpawnPlugin` is deprecated in v1.4, and will be removed in v1.6. "
-                "Notice that it will be overriden by the trainer setting."
-            )
-        self._num_nodes = num_nodes or 1
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:91]: this line needs to stay and be modified (see setter below) to 

`self._num_nodes = 1`

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 2]
```diff
@@ -83,18 +81,7 @@ def __init__(
...
-        if sync_batchnorm is not None:
-            rank_zero_deprecation(
-                "Argument `sync_batchnorm` in `DDPSpawnPlugin` is deprecated in v1.4, and will be removed in v1.6. "
-                "Notice that it will be overriden by the trainer setting."
-            )
-        self._sync_batchnorm = sync_batchnorm or False
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:97]: (similar comment as above)

**Comment by awaelchli** [comment]
@puhuk Great, thanks for working on this. A few items need to be resolved: 

1. See my review comments
2. Let's add a changelog entry in the "Removed" section 
3. You will also need to remove the two tests: `test_v1_6_0_ddp_spawn_num_nodes()` and `test_v1_6_0_ddp_spawn_sync_batchnorm()`

**Comment by awaelchli** [comment]
Seems to be a duplicate of  #10357

**Comment by puhuk** [comment]
@awaelchli 
It's different since this PR is about ddp_spawn.py, while #10357 to resolve ddp.py

**Pull request closed without merging**